### PR TITLE
C#: Make support for file scoped namespace declarations.

### PR DESCRIPTION
**Pull request by michaelnebel**

### CODE REVIEW

File: csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace MyFileScopedNamespace;
+public class MyFileScopedNamespaceClass
+{
+    private readonly object myField = new object();
+    public object MyProp { get; set; }
+    public void MyMethod() { }
+
```
Comment (hvitved) [inline csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs:7]: Remove

File: csharp/ql/test/library-tests/csharp10/fileScopedNamespace.ql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import csharp
+
+query predicate fileScopedNamespace(Namespace n, Member m) {
+  n.hasQualifiedName("MyFileScopedNamespace") and
+  exists(Class c |
+    c.getNamespace() = n and
+    c.hasMember(m) and
+    m.getLocation().toString().matches("%FileScopedNamespace.cs%")
+  )
+}
```
Comment (hvitved) [inline csharp/ql/test/library-tests/csharp10/fileScopedNamespace.ql:10]: I would also like to have
```ql
query predicate namespaceDeclaration(NamespaceDeclaration nd, Namespace n) { n = nd.getNamespace() }
```

File: csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+namespace MyFileScopedNamespace;
```
Comment (hvitved) [inline csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs:1]: I would prefer a newline after this line.

**Review by hvitved — COMMENTED** [review]
Looks great!

File: csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs (hunk 3 above)
Comment (michaelnebel) [reply csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs:1]: Yes, of course - VS code "cheated" me with the "reference" link (then it looks like there is an empty line unless you directly at the line numbers)

File: csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/test/library-tests/csharp10/FileScopedNamespace.cs:7]: Yes :-)

File: csharp/ql/test/library-tests/csharp10/fileScopedNamespace.ql (hunk 2 above)
Comment (michaelnebel) [reply csharp/ql/test/library-tests/csharp10/fileScopedNamespace.ql:10]: Yes that is a good idea!

**Comment by michaelnebel** [comment]
@hvitved : There are no changes since last time, but I added and removed and empty commit to re-trigger the CI.

**Review by hvitved — APPROVED** [review]

**Pull request merged**